test/functional/tool_wallet.py

```diff
@@ -390,7 +390,11 @@ def test_dump_createfromdump(self):
         bad_sum_wallet_dump = os.path.join(self.nodes[0].datadir, "wallet-bad_sum3.dump")
         dump_data["checksum"] = "2" * 10
         self.write_dump(dump_data, bad_sum_wallet_dump)
-        self.assert_raises_tool_error('Error: Dumpfile checksum does not match. Computed {}, expected {}{}'.format(checksum, "2" * 10, "0" * 54), '-wallet=badload', '-dumpfile={}'.format(bad_sum_wallet_dump), 'createfromdump')
+        self.assert_raises_tool_error('Error: Checksum is not the correct size', '-wallet=badload', '-dumpfile={}'.format(bad_sum_wallet_dump), 'createfromdump')
+        assert not os.path.isdir(os.path.join(self.nodes[0].datadir, "regtest/wallets", "badload"))
+        dump_data["checksum"] = "3" * 66
```

Comment:
My naive assumption was that the following change
```suggestion
        dump_data["checksum"] = "3" * 65
```
will keep test passing. But it is wrong.

Maybe verify the size of `value` before calling `ParseHex`?

Reply:
A hexadecimal string with an odd number of characters should cause a parse error anyway. Thinking of it, it's pretty bad that we don't have parse error feedback at all in `ParseHex`. Even if you'd add a non-hex character in between it's just break off the parsing early instead of return, say, nullopt.